### DISCUSSION TITLE
Controller needs also to invoke an external web service: use a minio?

### DISCUSSION THREAD

**Discussion by fluca1978** [mojolicious/mojo#2171] [Q&A]
I have a controller method that is going to store some data into the application database and render a page, so the usual.
Now, I have to add in the logic the fact that the controller method is going to call an external web service passing some data.

I don't like the idea of doing it at once before rendering the page to the user, so I'm guessing if using a minion (i.e., encoding for later execution) is a good option. I don't have to get immediate response, and using a queue will make it possible for several retries in the case the external endpoint fails.

Suggestions?

**Comment by kraih** [comment]
That's what job queues are meant for.
